File: Algorithms/p121_Best_Time_to_Buy_and_Sell_Stock_2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size()<=1){
            return 0;
        }
        
        int res = 0;
        int min = prices[0];
        for(int i=1;i<prices.size();++i){
            if(prices[i]>min){
                res = max(prices[i]-min,res);
            }else{
                min = prices[i];
            }
        }
        return res;
    }
};

File: Algorithms/p121_Best_Time_to_Buy_and_Sell_Stock_2.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices)<=1:
            return 0
        
        res = 0;
        min_price = prices[0]
        for i in range(1,len(prices)):
            if prices[i]>min_price:
                res = max(prices[i]-min_price,res)
            else:
                min_price = prices[i]
        return res
